lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -0,0 +1,411 @@
+    constant_5 = 2.0747825E-09
+    constant_6 = -9.484024E-13
+    constant_7 = 4.1635019
+    constant_8 = -5800.2206
+    constant_9 = 1.3914993
+    constant_10 = -0.048640239
+    constant_11 = 0.000041764768
+    constant_12 = -0.00000001445209310000
+    constant_13 = 6.5459673
+    temp = temperature.copy()
+    temp.convert_units('K')
+    check_range(temp, 173., 473.)
+    data = temp.data
+    for cell in np.nditer(data, op_flags=['readwrite']):
+        if cell <= 273.15:
+            cell[...] = (np.exp(constant_1 / cell + constant_2 + constant_3 *
+                                cell + constant_4 * cell ** 2 + constant_5 *
+                                cell ** 3 + constant_6 * cell ** 4 +
+                                constant_7 * np.log(cell))) / 1000
+        else:
+            cell[...] = (np.exp(constant_8 / cell + constant_9 + constant_10 *
+                                cell + constant_11 * cell ** 2 + constant_12 *
+                                cell ** 3 + constant_13 * np.log(cell))) / 1000
+    result = temp.copy(data=data)
+    result.units = Unit('kPa')
+    result.convert_units('Pa')
+    result.rename("saturated_vapour_pressure")
+    return result
+
+
+def saturation_vapour_pressure_goff_gratch(temperature, pressure):
+    '''
+    Saturation Vapour pressure calculation based on the
+    Goff-Gratch Equation (WMO standard method) and corrected for pressure.
+
+    Args:
+        temp : iris.cube.Cube
+            Cube of temperature which will be converted to Kelvin
+            prior to calculation
+            Valid from -100C to 200 C
+
+        pressure : iris.cube.Cube
+            Cube of pressure which will be converted to hectoPascals
+            prior to calculation
+
+    Returns:
+        svp : iris.cube.Cube
+            Cube containing the saturation vapour pressure of the
+            air in Pa
+
+    References:
+        Numerical data and functional relationships in science and technology.
+        New series. Group V. Volume 4. Meteorology. Subvolume b. Physical and
+        chemical properties of the air, P35.
+
+        Gill, Atmosphere-Ocean Dynamics, Appendix 4 Equation A4.7
+    '''
+
+    # Constants for vapour pressure over liquid water equation
+    constant1 = 10.79574
+    constant2 = 5.028
+    constant3 = 1.50475E-410000
+    constant4 = -8.2969
+    constant5 = 0.42873E-3
+    constant6 = 4.76955
+    constant7 = 0.78614
+    # Constants for vapour pressure over ice equation
+    constant8 = -9.09685
+    constant9 = 3.56654
+    constant10 = 0.87682
+    constant11 = 0.78614
+    triple_pt = constant.TRIPLE_PT_WATER
+
+    # Copy cubes to avoid modifying those passed to this function
+    # Check that units are correct for the calculations
+    press = pressure.copy()
+    press.convert_units('hPa')
+    temp = temperature.copy()
+    temp.convert_units('K')
+    check_range(temp, 173., 373.)
+
+    # create output cube
+    svp = temp.copy()
```

Comment:
This is technically okay because there is a temperature.copy() in the calling routine (which currently calls goff-gratch method, but could call this). However, the routine is overwriting the passed in temperature cube data in-situ. If someone were to call this routine as a standalone function with a temperature cube, they would find their input data modified. Maybe the cube.copy() that is going on in the calling function should be moved in to each of the saturated vapour pressure functions to avoid this issue. Any thoughts?